### ISSUE TITLE
new records’ hasMany relations won’t save on update/remove unless they are reloaded from server

### ISSUE THREAD

**Issue by williamli** [firebase/emberfire#96]
I have the following relations (async: true)

```
// city
districts hasMany district,
```
```
// district
city belongs to city
```

when a new district is store in a new city:
```
// new city
var city = this.store.createRecord('city');
city.save();
```
```
// new district in the city
var district = this.store.createRecord('district’);
district.set('city', city);
district.save().then(function(){
	Ember.debug('The district was saved.’);
        district.get('city').then(function(city){
              city.get('districts').then(function(districts){
	      districts.addObject(district);
              city.save().then(function(){
		Ember.debug('The district was linked to city.’);
….
….
```
In firebase, under the city, there is now a districts array

now if I try to remove district from city using:
```
// remove district
district.get('city').then(function (city){
	city.get('districts').then(function(districts){

		districts.forEach(function(d){… list all districts in the city… <— point A
		
		districts.removeObject(district);

		districts.forEach(function(d){… list all districts in the city… <— point B
			city.save().then(function(){
				Ember.debug('District unlinked from city.’);
…
```

in point A, all the district in the city will get printed

in point B, all the district in the city BUT the one i want to remove is listed (which is correct)

however, if city is just created, and saved, calling city.save() again won’t update the districts array…

if the city is loaded from firebase, the above code will remove the district from the districts array...

**Comment by tstirrat** [comment]
this should have been fixed in #206 (EmberFire v1.4.0+)

**Comment by tstirrat** [comment]
If this problem still persists, please make a [MCVE](http://stackoverflow.com/help/mcve) in a github repo (ember-cli) or jsbin/fiddle.
